Fc 776 requisition list documentation (#285)
front-commerce/developers.front-commerce.com: source/docs/appendices/migration-guides.md

 };
```

### `swatch` image format in base theme

> We backported the `swatch` image format from theme chocolatine to base theme. This image format is used to display product images of a requisition list. If you are using base theme and want to use the requisition list feature of the B2B module you have to add the swatch image format to your `src/config/images.js` as folows:

Reviewer: ```suggestion
We backported the `swatch` image format from theme chocolatine to base theme. This image format is used to display product images of a requisition list. If you are using the base theme and want to use the requisition list feature of the B2B module you have to add the swatch image format to your `src/config/images.js` as follows:
```

};
```

## Added requisition list features to B2B module (only theme chocolatine suported)

> If you are using base theme and wish to use B2B module you have to override `theme/modules/ProductView/Synthesis/AddProductToCart.js` from the base theme and provide a way to add to requisition list from the product page (see `<AddProductToRequisitionList>` and `modules/front-commerce-b2b/web/theme/modules/ProductView/Synthesis/AddProductToCart.js` for inspiration). You also need to override `theme/modules/AddToCart/_AddToCart.scss` if you have not already and just copy the styles from the base theme to it (`src/web/theme/modules/AddToCart/_AddToCart.scss`).

Reviewer: Can you maybe add links to the references?
- `<AddProductToRequisitionList>`
- `modules/front-commerce-b2b/web/theme/modules/ProductView/Synthesis/AddProductToCart.js`

Author: 0b9365cea93199a19913b3dd32e838f61dc983b9

> If you are using theme chocolatine and have overided `theme/modules/ProductView/Synthesis/AddProductToCart.js` and want to use the B2B module you need to apply the folowing change to it:

Reviewer 2: ```suggestion
If you are using theme chocolatine and have overridden `theme/modules/ProductView/Synthesis/AddProductToCart.js` and want to use the B2B module you need to apply the following change to it:
```

> If you are using base theme and wish to use B2B module you have to override `theme/modules/ProductView/Synthesis/AddProductToCart.js` from the base theme and provide a way to add to requisition list from the product page (see [AddProductToRequisitionList](https://gitlab.com/front-commerce/front-commerce/-/blob/2.12.0/modules/front-commerce-b2b/web/theme/modules/RequisitionList/AddProductToRequisitionList/AddProductToRequisitionList.js) and [B2B's AddProductToCart](https://gitlab.com/front-commerce/front-commerce/-/blob/2.12.0/modules/front-commerce-b2b/web/theme/modules/ProductView/Synthesis/AddProductToCart.js) for inspiration). You also need to override `theme/modules/AddToCart/_AddToCart.scss` if you have not already and just copy the styles from the base theme to it (`src/web/theme/modules/AddToCart/_AddToCart.scss`).

Reviewer 3: ```suggestion
If you are using base theme and wish to use the B2B module you have to override `theme/modules/ProductView/Synthesis/AddProductToCart.js` from the base theme and provide a way to add to requisition list from the product page (see [AddProductToRequisitionList](https://gitlab.com/front-commerce/front-commerce/-/blob/2.12.0/modules/front-commerce-b2b/web/theme/modules/RequisitionList/AddProductToRequisitionList/AddProductToRequisitionList.js) and [B2B's AddProductToCart](https://gitlab.com/front-commerce/front-commerce/-/blob/2.12.0/modules/front-commerce-b2b/web/theme/modules/ProductView/Synthesis/AddProductToCart.js) for inspiration). You also need to override `theme/modules/AddToCart/_AddToCart.scss` if you have not already and just copy the styles from the base theme to it (`src/web/theme/modules/AddToCart/_AddToCart.scss`).
```

+            children={({ match }) => (
+              <Link
+                to="/user/requisition-lists"
+                className={makeClassName(match, ordersCount === 0)}
+              >
+                {intl.formatMessage(messages.requisitionListsLink)}
+                <Icon icon="list" title="" />
+              </Link>
+            )}
+          />
+        )}
...
      </nav>
    );
  }
);
```
</details>
<br/>

#### New style sheets

If you have overriden `theme/components/_components.scss` you need to add the following line to it

```scss
@import "~theme/components/molecules/LoadingOverlay/LoadingOverlay";
@import "~theme/components/molecules/SelectMenu/SelectMenu";
```

If you are using the B2B module and have overriden `theme/_b2b.scss` you need to add the following lines to it

```scss
@import "~theme/components/organisms/Tree/Tree";
@import "~theme/modules/Company/CompanyStructure/CompanyStructure";
@import "~theme/modules/RequisitionList/RequisitionList";
@import "~theme/modules/RequisitionList/RequisitionListTable/RequisitionListTable";
```

If you overrode `src/web/theme/components/atoms/Button/_Button.scss` apply the following diff to it:

<details>

> <summary>Diff for base theme</strong></summary>

Reviewer 3: ```suggestion
<summary><strong>Diff for base theme</strong></summary>
```

```diff
...
+.button--disabled {
+  &:hover,
+  &:focus {
+    background: $white;
+  }
+}
...
.button--disabled {
+  &,
+  &:hover,
+  &:focus,
+  &:active {
+    background: $white;
+    text-decoration: none;
    cursor: not-allowed;
    border-color: $shade04;
    background: $white;
    color: $shade04;
    &.button--primary {
      border-color: $shade04;
      background: $shade04;
      color: $white;
    }
    &.button--icon {
      background: transparent;
      color: $shade04;
    }
+  }
}
```

</details>
<br/>

<details>

> <summary>Diff for theme chocolatine</strong></summary>

Reviewer 3: ```suggestion
<summary><strong>Diff for theme chocolatine</strong></summary>
```

```diff
...
.button--disabled {
  border-color: $shade03;
  &:hover,
- &:focus {
+ &:focus,
+ &:active {
    background: $white;
  }
}
...
.button--disabled {
-  cursor: not-allowed;
-
-  &,
-  &:hover {
+  &,
+  &:hover,
+  &:focus,
+  &:active {
+    text-decoration: none;
+    cursor: not-allowed;
    border-color: $shade03;
    background: $white;
    color: $shade03;
    &.button--primary {
      border-color: $shade03;
      background: $shade03;
      color: $fontColor;
    }
  }
}
```

</details>
<br/>

> if you overrode `theme/modules/AddToCart/_AddToCart.scss` and are using theme chocolatine apply the folowing diff to it:

Reviewer 2: ```suggestion
if you overrode `theme/modules/AddToCart/_AddToCart.scss` and are using theme chocolatine apply the following diff to it:
```
